src/monodroid/jni/monodroid-glue.cc

```diff
@@ -1864,11 +1864,24 @@ create_and_initialize_domain (JNIEnv* env, jclass runtimeClass, jstring_array_wr
 	return domain;
 }
 
+/* !DO NOT REMOVE! Used by the Android Designer */
 JNIEXPORT void JNICALL
 Java_mono_android_Runtime_init (JNIEnv *env, jclass klass, jstring lang, jobjectArray runtimeApksJava,
                                 jstring runtimeNativeLibDir, jobjectArray appDirs, jobject loader,
                                 jobjectArray externalStorageDirs, jobjectArray assembliesJava, jstring packageName,
                                 jint apiLevel, jobjectArray environmentVariables)
+{
+	Java_mono_android_Runtime_initInternal (
+		env, klass, lang, runtimeApksJava, runtimeNativeLibDir, 
+		appDirs, loader, externalStorageDirs, assembliesJava, apiLevel,
+		/* embeddedDSOsEnabled */ JNI_FALSE);
+}
+
+JNIEXPORT void JNICALL
+Java_mono_android_Runtime_initInternal (JNIEnv *env, jclass klass, jstring lang, jobjectArray runtimeApksJava,
+                                jstring runtimeNativeLibDir, jobjectArray appDirs, jobject loader,
+                                jobjectArray externalStorageDirs, jobjectArray assembliesJava,
+                                jint apiLevel, jboolean embeddedDSOsEnabled)
```

Comment:
I thought this "overload" was reasonable so we can change these parameters. Is that a good idea?

Reply:
It does give us some leeway, I like it.